File: csharp/src/RespBench/Command/Impl/PingCommand.cs
/*
 * Copyright 2025 the original author or authors.
 */
using RespBench.Client;
using RespBench.Config;
using RespBench.Engine;

namespace RespBench.Command.Impl;

public class PingCommand : ICommand
{
    private double _weight = 1.0;

    public string Name => "PING";
    public string Description => "PING [message]";
    public double Weight => _weight;

    public void Configure(CommandConfig config)
    {
        _weight = config.Weight;
    }

    public async Task<CommandResult> Execute(IBenchmarkClient client, KeyGenerator keyGenerator)
    {
        try
        {
            var result = await client.Ping().ConfigureAwait(false);
            return CommandResult.SuccessResult("PING", result.LatencyMicros);
        }
        catch (Exception e)
        {
            return CommandResult.Failure("PING", 0, e.Message);
        }
    }
}

File: php/src/Command/Impl/PingCommand.php
<?php

declare(strict_types=1);

namespace RespBench\Command\Impl;

use RespBench\Client\BenchmarkClient;
use RespBench\Command\Command;
use RespBench\Command\CommandResult;
use RespBench\Engine\KeyGenerator;

final class PingCommand extends Command
{
    public function execute(BenchmarkClient $client, KeyGenerator $keyGenerator): CommandResult
    {
        $result = $client->ping();

        return new CommandResult($this->name, $result->latencyMicros, $result->isSuccess());
    }
}
